.gitignore

```diff
@@ -16,13 +16,55 @@ testbin/
 /webrev/
 /webrev.zip
 
+# IntelliJ Idea Begin
+# https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore
 # Ignore IntelliJ files
-.idea/tasks.xml
-.idea/workspace.xml
 .idea/inspectionProfiles/**
 .idea/copyright/profiles_settings.xml
-.idea/codeStyles/**
-.idea/checkstyle-idea.xml
+
+# Code styles should be shared
+#.idea/codeStyles/**
+#.idea/checkstyle-idea.xml
+
+# User-specific stuff
+.idea/**/workspace.xml
+.idea/**/tasks.xml
+.idea/**/usage.statistics.xml
+.idea/**/dictionaries
+.idea/**/shelf
+
+# Generated files
+.idea/**/contentModel.xml
+
+# Gradle
+.idea/**/gradle.xml
+.idea/**/libraries
+
+# Auto-import
+.idea/artifacts
+.idea/compiler.xml
+.idea/jarRepositories.xml
+.idea/modules.xml
+.idea/*.iml
+.idea/modules
+*.iml
+*.ipr
+
+# File-based project format
+*.iws
+
+# IntelliJ
+out/
+
+# SonarLint plugin
+.idea/sonarlint/
+#IntelliJ Idea End
+
+#Crash log
+hs_err*.log,
+
+#mac folder
+.DS_Store
```

Comment:
These seem unrelated to this PR (and in the case of `hs_err`, there is a typo).

Reply:
Removed them